packages/babel-parser/src/parser/expression.js

```diff
@@ -653,8 +651,10 @@ export default class ExpressionParser extends LValParser {
     refExpressionErrors: ?ExpressionErrors,
   ): N.Expression {
     if (update) {
-      this.checkLVal(node.argument, "prefix operation");
-      return this.finishNode(node, "UpdateExpression");
+      this.checkLVal(node.argument, {
+        in: (node = this.finishNode(node, "UpdateExpression")),
```

Comment:
```suggestion
        in: this.finishNode(node, "UpdateExpression"),
```
it already mutates `node`.